### PR TITLE
 added polish translation 

### PR DESCRIPTION
**Pull request by dipekthebun**
I am stupid and my previous pull request wasnt merging with dev (really sorry).

I was informed that the translation system will be changing so I will mark this pull request as a draft.

https://fontstruct.com/fontstructions/show/1765530/pixellocale-v-1-4
I am leaving this link to the font as a mental note so I can add it when the translation is fiinished with the new system and strings.

### CODE REVIEW

**Comment by OrsellGaming** [comment]
Hey @dipekthebun! We have finally got dev stuff synced and merged! Feel free to sync your fork of the repo and make any changes you need. We'll look through the pull request once you've finished your part.

**Comment by dipekthebun** [comment]
Hi, the pull request is ready now, although I am not sure how to add the font that I mentioned in the request. is it possible to add a font specifically for a language?

**Comment by cabiste69** [comment]
yeah just add a property in the json `"font"`, write the name of the font there then include the font in your commit

**Comment by OrsellGaming** [comment]
Hey @dipekthebun. In pull request #56, the requester has changed up how translations work and have added some extra translations. Once we merge that pull request you'll be able to sync and pull everything with dev branch again for the final translations on your end. From there once you're finished and ready we can merge your pull request.

**Comment by OrsellGaming** [comment]
Hey @dipekthebun! Is it possible for you to do a translation for the `README.md` file? It's possible to make one that just appears for just those who have polish set as their primary language by naming the file `README.(the polish two letter language code).md`. It would be greatly appreciated!

**Comment by dipekthebun** [comment]
@cabiste69 
What do you mean by "write the name of the font there then include the font in your commit"?
Should it be just the filename or the path?

**Comment by OrsellGaming** [comment]
@dipekthebun Basically if you have a specific font for the polish language, because not all languages support the pixel font, you add the `font` JSON key at the top of the file, then specify the path to the where the supported font is in the launchers GUI font folder.

**Comment by dipekthebun** [comment]
Hi, im not very experienced with the launcher. Since I added the "font" key to the file it doesnt seem like its loading the language.
I checked the logs and it says this

```
searching for available languages
Language set: Polski
Finding config path...
Deleted old file
Writing to file...
Finding config path...
Deleted old file
Writing to file...
[ERROR] language file isn't found or key mismatch
Language Set!
```
Any help?

**Comment by cabiste69** [comment]
Hey sorry about the late response, basically if you're using a font that's different thsn the original one then write in the polish.json file `"font" : "fontname"` and replace the `fontname` with the name of the font you're using
then in the put the file of the font in `gui/assets/fonts` that should be all

**Comment by OrsellGaming** [comment]
> Hi, im not very experienced with the launcher. Since I added the "font" key to the file it doesnt seem like its loading the language. I checked the logs and it says this
> 
> ```
> searching for available languages
> Language set: Polski
> Finding config path...
> Deleted old file
> Writing to file...
> Finding config path...
> Deleted old file
> Writing to file...
> [ERROR] language file isn't found or key mismatch
> Language Set!
> ```
> 
> Any help?

can you send a snippet of the JSON `font` key? ill see if something is wrong with that

**Comment by dipekthebun** [comment]
@OrsellGaming 
```
{
  "font": "pixellocale.ttf",
```

**Comment by OrsellGaming** [comment]
> @OrsellGaming
> 
> ```
> {
>   "font": "pixellocale.ttf",
> ```

Ok so the issue is that the path doesn't target the font directly, it should be like `"font": "src/GUI/assets/fonts/pixel.ttf"`. This way the launcher can grab the full path to the font.

**Comment by dipekthebun** [comment]
@OrsellGaming Sorry for the late response, I tried entering the entire path yet the same problem remains, it doesnt load.

**Comment by OrsellGaming** [comment]
@dipekthebun I think I know the issue but you need to give me a bit to fix it.

**Comment by OrsellGaming** [comment]
@dipekthebun I think I know the issue but you will need to give me a bit to fix the issue. Then it should work after you pull and sync the change on your repo. We also added a slightly different system for detecting the systems language on start. Here is a picture of the detection in the Configs.py file. You should be able to add a detection for the Polish translation, what is returned is the name of the translation file without the ".json" extension. Essentually it will detect the two letter language code for the system and return based on that code, so make sure to stick in the language codes for Polish.
image

**Comment by OrsellGaming** [comment]
Hey @dipekthebun, if you sync and pull changes to your repo fork, you should be able to get the fixes that will allow you to use the polish translation.

**Comment by OrsellGaming** [comment]
Hey @dipekthebun! Is everything working correctly with the Polish translation? If you are done and ready, then I will be able to merge your pull request.

**Comment by dipekthebun** [comment]
@OrsellGaming Hi very sorry but the translation is not fully finished yet. I am going to finish it later this day.

File: src/languages/English.json [hunk 1]
```diff
@@ -31,7 +31,10 @@
   "game_files_no_cached_files": "There are no cached files for the mod!\n\nThe latest version of the ModFiles folder will be downloaded...",
   "workshop_menu_info": "The Workshop Menu is for getting the change level\ncommand to change to a certain workshop map.\n\nMake sure you have the workshop map downloaded\nfirst before running the command in the Portal 2 console!",
   "workshop_link": "Enter a Steam workshop map link:",
-  "workshop_map_not_found": "Could not find workshop map!\nMake sure you are subscribed\nto the map and have played it\nat least once.",
+  "workshop_map_not_found": "Nie znaleziono mapy",
+  "workshop_sentence0_sure-you-are-sub": "Upewnij się że zasubskrybowałeś",
+  "workshop_sentence0_to-map-and-play-it": "mapę i zagrałeś w nią",
+  "workshop_sentence0_least-once": "przynajmniej jeden raz.",
```
Comment (cabiste69) [inline src/languages/English.json:37]: uhm... i think you mistook the english file for the polish file

Comment (dipekthebun) [reply src/languages/English.json]: Yeah, no clue how I made that mistake! I fixed it.

**Comment by dipekthebun** [comment]
Hi, I am currently busy with schoolwork so I won't be able to finish the strings until the weekend! Sorry about that

**Comment by cabiste69** [comment]
It's all good man take your time and sorry if we complicated it for you

**Comment by dipekthebun** [comment]
I dont know why the pull request closed, reopening it now

**Comment by cabiste69** [comment]
well you just deleted all my work by mistake 💀

**Comment by dipekthebun** [comment]
I presume that line 98 in the english .json is supposed to say "Auto-unmount"?
In line 122 of english.json there are 2 typos, "joinning" and "rich presense" (it should be "rich presence")

> well you just deleted all my work by mistake 💀
what do you mean?
It seems that my fork is on par with the main dev branch?

**Comment by dipekthebun** [comment]
the translation is done yet i cant seem to run the dev branch build due to the datasystem module missing.
is the branch not ready yet or am i doing something wrong?

**Comment by cabiste69** [comment]
> the translation is done but i cant test it because i cant seem to run the dev branch build due to the datasystem module missing. is the branch not ready yet or am i doing something wrong?

as i told you earlier, you overwrote my code that fixed the launcher

you basically had a merge conflict and you force pushed it (deleted all the stuff that was different from yours)

**Comment by dipekthebun** [comment]
> > the translation is done but i cant test it because i cant seem to run the dev branch build due to the datasystem module missing. is the branch not ready yet or am i doing something wrong?
> 
> as i told you earlier, you overwrote my code that fixed the launcher
> 
> you basically had a merge conflict and you force pushed it (deleted all the stuff that was different from yours)

oh shit my bad gonna fix it now

**Comment by dipekthebun** [comment]
it seems that it all works now!
if someone can review me then please do!

**Comment by cabiste69** [comment]
this pr would break the launcher if it was merged

**Pull request closed without merging**